packages/backend/src/controllers/authentication/strategies/slackStrategy.ts

```diff
@@ -0,0 +1,71 @@
+/// <reference path="../../../@types/passport-slack-oauth2.d.ts" />
+import {
+    getErrorMessage,
+    OpenIdIdentityIssuerType,
+    OpenIdUser,
+    UnexpectedDatabaseError,
+} from '@lightdash/common';
+import { Strategy as SlackStrategy } from 'passport-slack-oauth2';
+import { lightdashConfig } from '../../../config/lightdashConfig';
+
+export const slackPassportStrategy = !(
+    lightdashConfig.slack?.clientId && lightdashConfig.slack?.clientSecret
+)
+    ? undefined
+    : new SlackStrategy(
+          {
+              scope: ['openid'],
+              clientID: lightdashConfig.slack.clientId,
+              clientSecret: lightdashConfig.slack.clientSecret,
+              callbackURL: `${lightdashConfig.siteUrl.replace(
+                  'http://',
+                  'https://',
+              )}/api/v1/auth/slack/callback`,
+              state: true,
+              passReqToCallback: true,
+          },
+          async (req, accessToken, refreshToken, profile, done) => {
+              if (!req.user || req.user.email === undefined) {
+                  return done(new Error('Invalid user'));
+              }
```

Comment:
Edge case here:

1. The user is not logged into Lightdash on their browser.
2. They prompt the AI on Slack.
3. They receive a link for `/api/v1/auth/slack`.
4. When they click the link, they encounter an error 500.

Currently, we lack the logic to save the user’s state (url) and redirect them back to it after logging in. Therefore, we can consider this a known limitation unless you have any suggestions.

We could just redirect to login, so that they can log-in and then go to slack and try again

Reply:
good point, I'll take alook

Reply:
Fixed by adding a redirection to /login?redirect=/auth/slack 
if user is not authenticated.

[Screencast from 2025-07-04 16-36-27.webm](https://github.com/user-attachments/assets/eee5519d-e7ed-4e9d-80e6-394ff197459d)